src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultClaimUidExtractor.cs

```diff
@@ -22,42 +22,89 @@ public DefaultClaimUidExtractor(ObjectPool<AntiforgerySerializationContext> pool
         }
 
         /// <inheritdoc />
-        public string ExtractClaimUid(ClaimsIdentity claimsIdentity)
+        public string ExtractClaimUid(IEnumerable<ClaimsIdentity> claimsIdentities)
         {
-            if (claimsIdentity == null || !claimsIdentity.IsAuthenticated)
+            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsIdentities);
+            if (uniqueIdentifierParameters == null)
             {
                 // Skip anonymous users
                 return null;
             }
 
-            var uniqueIdentifierParameters = GetUniqueIdentifierParameters(claimsIdentity);
             var claimUidBytes = ComputeSha256(uniqueIdentifierParameters);
             return Convert.ToBase64String(claimUidBytes);
         }
 
         // Internal for testing
-        internal static IEnumerable<string> GetUniqueIdentifierParameters(ClaimsIdentity claimsIdentity)
+        internal static IEnumerable<string> GetUniqueIdentifierParameters(IEnumerable<ClaimsIdentity> claimsIdentities)
         {
-            var nameIdentifierClaim = claimsIdentity.FindFirst(
-                claim => string.Equals(ClaimTypes.NameIdentifier, claim.Type, StringComparison.Ordinal));
-            if (nameIdentifierClaim != null && !string.IsNullOrEmpty(nameIdentifierClaim.Value))
+            if (claimsIdentities == null)
```

Comment:
Actually, this isn't a real scenario since the caller checks exactly this condition. Just `Debug.Assert()`.